// ruucm-util/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
/**
 * Usage
 */
// import util from '../../../../../ruucm-util/';
// import { log } from '../../../../../ruucm-util/';

var environment = {
  production: false
};
var log = exports.log = function log(name, data) {
  if (data === void 0) {
    data = null;
  }

  if (!environment.production) {
    if (data) {
      console.log(name + " ->", data);
      if (data === false || data === true || data === null || data === undefined) {
        console.log(data);
      }
    } else console.log(name);
  }
};
var objectValue = exports.objectValue = function objectValue(func, defaultValue) {
  try {
    return func();
  } catch (err) {
    return defaultValue;
  }
};
var ArrayToObject = exports.ArrayToObject = function ArrayToObject(arr) {
  var rv = {};
  for (var i = 0; i < arr.length; ++i) {
    rv[arr[i]] = true;
  }return rv;
};
var ObjectToArray = exports.ObjectToArray = function ObjectToArray(obj) {
  var result = Object.values(obj);
  return result;
};

var commaPipe = exports.commaPipe = function commaPipe(number) {
  if (number != null) return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
};

var getParameterByName = exports.getParameterByName = function getParameterByName(name, url) {
  if (!url) url = window.location.href;
  name = name.replace(/[\[\]]/g, "\\$&");
  var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
      results = regex.exec(url);
  if (!results) return null;
  if (!results[2]) return "";
  return decodeURIComponent(results[2].replace(/\+/g, " "));
};

exports.default = {
  log: log,
  getParameterByName: getParameterByName,
  objectValue: objectValue,
  ArrayToObject: ArrayToObject,
  ObjectToArray: ObjectToArray
};
