// @framer/ruucm.html-video/code/HTML_Video.tsx
import * as React from 'react'
import { PropertyControls, ControlType } from 'framer'
import styled from 'styled-components'

const Wrap = styled.div``
const Video = styled.video`
  width: 100%;
  height: 100%;
`

// Define type of property
interface Props {
  src: string
  playerState: string
  autoplay: boolean
  controls: boolean
  loop: boolean
  muted: boolean
}

export class HTML_Video extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    src: 'https://www.w3schools.com/html/mov_bbb.mp4',
    width: 791,
    height: 445,
    autoplay: false,
    controls: false,
    loop: true,
    muted: true,
  }

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    src: { type: ControlType.String, title: 'Source' },
    autoplay: { type: ControlType.Boolean, title: 'autoplay' },
    controls: { type: ControlType.Boolean, title: 'controls' },
    loop: { type: ControlType.Boolean, title: 'loop' },
    muted: { type: ControlType.Boolean, title: 'muted' },
  }

  play(myVideo) {
    myVideo.play()
  }
  pause(myVideo) {
    myVideo.pause()
  }
  stop(myVideo) {
    myVideo.load()
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.playerState != nextProps.playerState) {
      var myVideo = document.getElementById('video1')
      switch (nextProps.playerState) {
        case 'play':
          this.play(myVideo)
          break
        case 'pause':
          this.pause(myVideo)
          break
        case 'stop':
          this.stop(myVideo)
          break

        default:
          break
      }
    }
  }

  render() {
    return (
      <Wrap>
        <Video
          id="video1"
          autoplay={this.props.autoplay}
          controls={this.props.controls}
          loop={this.props.loop}
          muted={this.props.muted}
        >
          <source src={this.props.src} type="video/mp4" />
        </Video>
      </Wrap>
    )
  }
}
